src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImpl.java

```diff
@@ -1,46 +1,93 @@
 package uk.gov.companieshouse.api.accounts.service.impl;
 
-import java.security.NoSuchAlgorithmException;
 import java.security.SecureRandom;
 import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.beans.factory.annotation.Qualifier;
-import org.springframework.data.mongodb.repository.MongoRepository;
 import org.springframework.stereotype.Service;
-import uk.gov.companieshouse.api.accounts.Kind;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+import uk.gov.companieshouse.GenerateEtagUtil;
+import uk.gov.companieshouse.api.accounts.AttributeName;
+import uk.gov.companieshouse.api.accounts.LinkType;
 import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountEntity;
 import uk.gov.companieshouse.api.accounts.model.rest.CompanyAccount;
+import uk.gov.companieshouse.api.accounts.repository.CompanyAccountRepository;
 import uk.gov.companieshouse.api.accounts.service.CompanyAccountService;
-import uk.gov.companieshouse.api.accounts.transformer.GenericTransformer;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.transaction.TransactionManager;
+import uk.gov.companieshouse.api.accounts.transformer.CompanyAccountTransformer;
 
 @Service
-public class CompanyAccountServiceImpl extends
-        AbstractServiceImpl<CompanyAccount, CompanyAccountEntity> implements CompanyAccountService {
+public class CompanyAccountServiceImpl implements CompanyAccountService {
 
     @Autowired
-    public CompanyAccountServiceImpl(
-            @Qualifier("companyAccountRepository") MongoRepository<CompanyAccountEntity, String> mongoRepository,
-            @Qualifier("companyAccountTransformer") GenericTransformer<CompanyAccount, CompanyAccountEntity> transformer) {
-        super(mongoRepository, transformer);
+    private TransactionManager transactionManager;
+
+    @Autowired
+    private CompanyAccountRepository companyAccountRepository;
+
+    @Autowired
+    private CompanyAccountTransformer companyAccountTransformer;
+
+    /**
+     * {@inheritDoc}
+     */
+    public CompanyAccount createCompanyAccount(CompanyAccount companyAccount) {
```

Comment:
we should be passing the request into this service so it is available to get the transaction and for use in logging for following requests through the service

Reply:
I think it would be bad practice to pass the request into the service layer, we should maintain a separation of concerns by extracting what we need and passing that instead.

Reply:
That depends on which was we go with the return value and exception handling from service back to controller (comment above).

If we use typed exceptions to determine response codes in the controller then the service knows nothing about the request so thats fine - the only issue then is that the structured logging will not have access to the request for tracing it through the service.
Personally I'd prefer useful logs that allow us to trace requests through the controller and service.

Reply:
We shouldn't be using the Structured logging lib at all in place of an established industry standard logging framework, but that aside if Structured logging needs something in the service layer from the request it should be extracted and passed in.

Reply:
We have to output structured JSON logs from our microservices and be able to trace requests through all of our services using the request ID from headers.

If there is an industry standard way of doing this without using our structured-logging library (I'm sure there will actually be many ways of doing this) then we can look at using it instead of our structured-logging library.
Maybe worth a look into options going forwards but, as of right now, the structured-logging library is the library that all of our java microservices use, including this one.

As a minimum to get this functionality with the current set up we need to pass in the requestID (rather than the entire request object) to the service so that it can use it for traceability through the logs.

Reply:
The Structured logging lib is effectively just System.out.println and there is a reason the industry considers it unsafe or poor practice to introduce System.out into production code. I know the volumes of traffic we are expecting is low in general terms and that it provides the benefit of consistency in logging but we should still be very aware of the fundamental issues we are introducing referencing System in a production environment.

Reply:
Agreed, logging to System.out isn't ideal and we should look at improving our structured logging functionality to avoid doing that whether that means changing the underlying implementation in our structured-logging library or moving to something else that gives us the same functionality. 
As I said previously if there is a better industry standard that gives us the same functionality then we can look at switching, if you have any proposals for this we can go through it offline and bring the relevant people into the conversation.